### PR TITLE
Install wxPython and make the ELF executable

### PR DESCRIPTION
**Pull request by cclauss**
This pull request pip installs wxPython before pyinstaller creates the Linux ELF file.
It also sets the execution permission on the ELF file so that it can be run.

### CODE REVIEW

**Comment by bittner** [comment]
Please, always provide a motivation for the PR! An explanatory comment, to make it easier to review (understand) the changes.

I'll do this once again on your behalf, if you don't mind.

**Comment by cclauss** [comment]
Done. Above.

**Comment by bittner** [comment]
References:

- [Description of issue](https://github.com/PythonTurtle/PythonTurtle/pull/151#issuecomment-698645807) solved with this PR
- [Current build process for a Linux bundle](https://github.com/PythonTurtle/PythonTurtle/pull/149/files)
- [Installation instructions](https://github.com/PythonTurtle/PythonTurtle#installation) (README)
- [Original build process on Travis CI](https://github.com/PythonTurtle/PythonTurtle/blob/master/.travis.yml#L65-L72)

File: .github/workflows/build_linux_elf.yml [hunk 1]
```diff
@@ -13,7 +13,11 @@ jobs:
     - uses: actions/setup-python@v2
     - run: sudo apt-get install libsdl2-2.0-0
     - run: python -m pip install pyinstaller
+    - env:
+        URL: https://extras.wxpython.org/wxPython4/extras/linux/gtk3/ubuntu-18.04
+      run: python -m pip install --find-links ${URL} wxPython
     - run: python setup.py clean bundle
+    - run: chmod +x /home/runner/work/PythonTurtle/PythonTurtle/dist/PythonTurtle
```
Comment (bittner) [inline .github/workflows/build_linux_elf.yml:20]: I took a look at the documentation. The [permission will be lost](https://github.com/marketplace/actions/upload-a-build-artifact#permission-loss)  in any case. The only solution is to [tar up the file](https://github.com/marketplace/actions/upload-a-build-artifact#maintaining-file-permissions-and-case-sensitive-files) instead.

All in all, I'm not happy with the zipp'ed binary. It should be possible to download the binary directly. Also, the [Upload-Artifact](https://github.com/marketplace/actions/upload-a-build-artifact) is probably the wrong Action. There should be a [Create-Release](https://github.com/actions/create-release) action that does what we need, instead.

**Review by bittner — CHANGES_REQUESTED** [review]

**Comment by bittner** [comment]
I'm happy you continue on this road to get the implementation into a final state.

Note that this can (should!) only be an intermediate solution. It's because where people would look for downloads is always the repository's [Releases](https://github.com/PythonTurtle/PythonTurtle/releases) section. We need to publish new version there, in the long run.

I was afraid I had pointed you into the current direction from the beginning. But looking at [our past conversations](https://github.com/PythonTurtle/PythonTurtle/issues/107#issuecomment-696251476) I see that I suggested using the `Create-Release` GitHub Action (over the `Upload-Artifact`) from the beginning. – I believe, uploading artifacts really has a different use case than publishing releases. It's more for intermediate results you want to use in later stages of automation, or asynchronous pipeline processes.

File: .github/workflows/build_linux_elf.yml [hunk 2]
```diff
@@ -18,7 +18,8 @@ jobs:
       run: python -m pip install --find-links ${URL} wxPython
     - run: python setup.py clean bundle
     - run: chmod +x /home/runner/work/PythonTurtle/PythonTurtle/dist/PythonTurtle
+    - run: tar -cvf PythonTurtle_for_Linux.tar /home/runner/work/PythonTurtle/PythonTurtle/dist/PythonTurtle
```
Comment (bittner) [inline .github/workflows/build_linux_elf.yml:21]: With the `z` option in addition yop can turn the result into a GZipped Tar archive, e.g.
```suggestion
    - run: tar -cfvz PythonTurtle_for_Linux.tgz /home/runner/work/PythonTurtle/PythonTurtle/dist/PythonTurtle
```

**Comment by cclauss** [comment]
> I suggested using the `Create-Release` GitHub Action (over the `Upload-Artifact`) from the beginning.

Yes.  I agree but I have taken a couple of shots at using `Create-Release` without success.  Perhaps if you or someone else could create a PR to demonstrate how `Create-Release` would be applied to this project, then I could plug my efforts into that.  Sorry for being so dense.

**Comment by bittner** [comment]
Have you tried the [code sample in their README](https://github.com/actions/create-release#example-workflow---create-a-release)? It should be pretty much copy+paste. Not even configuring a secret seems to be needed:

```yaml
on:
  push:
    tags:
    - "*"

jobs:
  deploy:
    runs-on: ubuntu-latest
    steps:
      - name: Create Release
        id: create_release
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          tag_name: ${{ github.ref }}
          release_name: PythonTurtle-v${{ github.ref }}
```

(I omitted the `body`, `draft` and `prerelease` options, assuming they are empty or `false` by default.)

**Comment by bittner** [comment]
It appears that creating the build artifacts and releasing are (meant to be) separate activities.

If that is true, the release workflow may look like this:

1. Every time a **change is merged to `master`** we run the builds that **create the artifacts** (binaries for all target platforms)
1. If a Git **tag is pushed to `master`** then the create release workflow uses the artifact of builds related to that very commit to **craft a release**.

Now, how do we access the artifacts of the `${{ github.ref }}` and feed them into the `Create-Release` Action? There are some interesting examples [in the issues](https://github.com/actions/create-release/issues) of their repository, but I didn't yet understand how supplying the artifacts is done. :thinking:

**Comment by cclauss** [comment]
One workflow [`needs`](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#creating-dependent-jobs) other workflows?

**Comment by bittner** [comment]
Good point! I hope that doesn't tightly couple the two workflows. (Imagine you have to directly create a release, but you wanted to - manually - try out the binaries first. That would be possible with independent workflows.)

For the assets, it seems like one must first create a release and then [upload assets](https://developer.github.com/v3/repos/releases/#upload-a-release-asset) using the `upload_url` value of the response of the create release API endpoint. This is also hinted at [from their README](https://github.com/actions/create-release#outputs), I figured. Looks like we're getting closer! :crossed_fingers:

**Comment by bittner** [comment]
Issue https://github.com/actions/create-release/issues/79 seems to describe a setup that comes close to what we probably need.

**Comment by cool-RR** [comment]
Me walking into this thread like:

**Pull request closed without merging**